frontend/src/resources/utils/get-cluster.ts

```diff
@@ -584,8 +584,41 @@ export function getDistributionInfo(
             break
     }
 
+    const versionRegex = /([\d]{1,5})\.([\d]{1,5})\.([\d]{1,5})/
+
+    function isVersionGreater(versionX: string, versionY: string) {
+        const matchesA = versionX.match(versionRegex)
+        const matchesB = versionY.match(versionRegex)
+        if (matchesA && matchesB && matchesA.length === 4 && matchesB.length === 4) {
+            if (parseInt(matchesA[1], 10) > parseInt(matchesB[1], 10)) {
+                return true
+            } else if (parseInt(matchesA[1], 10) === parseInt(matchesB[1], 10)) {
+                if (parseInt(matchesA[2], 10) > parseInt(matchesB[2], 10)) {
+                    return true
+                } else if (parseInt(matchesA[2], 10) === parseInt(matchesB[2], 10)) {
+                    if (parseInt(matchesA[3], 10) > parseInt(matchesB[3], 10)) {
+                        return true
+                    }
+                }
+            }
+            return false
+        }
+    }
+
+    function isCuratorUpgradeVersionValid() {
+        const currentVersionMCI = managedClusterInfo?.status?.distributionInfo?.ocp.version
+        const desiredVersionMCI = managedClusterInfo?.status?.distributionInfo?.ocp.desiredVersion
```

Comment:
Yes, `desiredVersion` is deprecated. See line 658 for how to handle. Also, does this need to be a function?